base/loading.jl

```diff
@@ -1210,7 +1214,11 @@ function compilecache(pkg::PkgId, path::String)
     end
     # run the expression and cache the result
     verbosity = isinteractive() ? CoreLogging.Info : CoreLogging.Debug
+    isinteractive() && precompiling_interactively!(true)
     @logmsg verbosity "Precompiling $pkg"
+    precomp_msg = " ► $(pkg.name)"
+    should_log_deps = is_precompiling_interactively() && !isinteractive() && (CoreLogging.current_logstate().min_enabled_level > CoreLogging.Debug) #don't print for main package or when in debug
```

Comment:
If you want to use the logging system rather than `print` that's great.  But an ad-hoc mixing some use of `CoreLogging` internals with the use of `print` is not the way to go!

What does this use of `min_enabled_level` achieve in practice?  Precompilation always happens in a separate process, so I don't think you can easily affect `min_enabled_level` in that process.

Reply:
I assumed that under the current behavior, the precomp subprocesses would adopt the main `JULIA_DEBUG` state, so if the user requests debug logging, the deps would print debug precomp messages.. but I didn't check that. 

I intended to preserve that behavior without messing up the messages with overprinting.